#!/bin/bash
#PBS -l mem=100gb,nodes=1:ppn=1,walltime=72:00:00
#PBS -A hirschc1
#PBS -m abe
#PBS -o /home/hirschc1/oconnorc/OandE
#PBS -e /home/hirschc1/oconnorc/OandE
#PBS -M oconnorc@umn.edu
#PBS -q mangi
#PBS -N fb_process

newgrp - hirschc1

#job to run samtools filter and sort using gnu parallel

#go here: https://www.msi.umn.edu/support/faq/how-can-i-use-gnu-parallel-run-lot-commands-parallel

#load needed modules
module load parallel
module load gatk/4.1.2
module load java/jdk1.8.0_171

cd /home/hirschc1/oconnorc/software/GenomeAnalysisTK-3.8-1-0-gf15c1c3ef
#Mo17 needs to be redone
#java -cp GenomeAnalysisTK.jar org.broadinstitute.gatk.tools.CatVariants -R /home/hirschc1/oconnorc/maize_refs/Mo17_chr1-10.fasta -V /home/hirschc1/oconnorc/freebayes_output/WiDiv512_Mo17_chr1A_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv512_Mo17_chr1B_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv512_Mo17_chr1C_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv512_Mo17_chr1D_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv512_Mo17_chr2A_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv512_Mo17_chr2B_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv512_Mo17_chr2C_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv512_Mo17_chr3A_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv512_Mo17_chr3B_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv512_Mo17_chr3C_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv512_Mo17_chr4A_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv512_Mo17_chr4B_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv512_Mo17_chr4C_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv512_Mo17_chr5A_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv512_Mo17_chr5B_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv512_Mo17_chr5C_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv512_Mo17_chr6A_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv512_Mo17_chr6B_gene1kb.vcf  -V /home/hirschc1/oconnorc/freebayes_output/WiDiv512_Mo17_chr7A_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv512_Mo17_chr7B_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv512_Mo17_chr8A_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv512_Mo17_chr8B_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv512_Mo17_chr8C_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv512_Mo17_chr9A_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv512_Mo17_chr9B_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv512_Mo17_chr10A_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv512_Mo17_chr10B_gene1kb.vcf -out /home/hirschc1/oconnorc/freebayes_output/WiDiv512_Mo17_allchr_gene1kb.vcf 

#java -cp GenomeAnalysisTK.jar org.broadinstitute.gatk.tools.CatVariants -R /home/hirschc1/oconnorc/maize_refs/PHB47_chr1-10.fasta -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_chr1A_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_chr1B.1_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_chr1B.2_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_chr1C_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_chr1D_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_chr2A_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_chr2B_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_chr2C_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_chr3A_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_chr3B_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_chr3C_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_chr4A_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_chr4B_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_chr4C_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_chr5A_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_chr5B1_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_chr5B2_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_chr5C_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_chr6A_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_chr6B_gene1kb.vcf  -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_chr7A_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_chr7B_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_chr8A_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_chr8B_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_chr8C_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_chr9A_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_chr9B_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_chr10A_gene1kb.vcf -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_chr10B_gene1kb.vcf -out /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_allchr_gene1kb.vcf

#java -cp GenomeAnalysisTK.jar org.broadinstitute.gatk.tools.CatVariants -R ~/maize_refs/PH207_chr1-10.fasta -V ~/freebayes_output/WiDiv510_PH207_chr1A_gene1kb.vcf -V ~/freebayes_output/WiDiv510_PH207_chr1B_gene1kb.vcf -V ~/freebayes_output/WiDiv510_PH207_chr1C_gene1kb.vcf -V ~/freebayes_output/WiDiv510_PH207_chr1D_gene1kb.vcf -V ~/freebayes_output/WiDiv510_PH207_chr2A_gene1kb.vcf -V ~/freebayes_output/WiDiv510_PH207_chr2B_gene1kb.vcf -V ~/freebayes_output/WiDiv510_PH207_chr2C_gene1kb.vcf -V ~/freebayes_output/WiDiv510_PH207_chr3A_gene1kb.vcf -V ~/freebayes_output/WiDiv510_PH207_chr3B_gene1kb.vcf -V ~/freebayes_output/WiDiv510_PH207_chr3C_gene1kb.vcf -V ~/freebayes_output/WiDiv510_PH207_chr4A_gene1kb.vcf -V ~/freebayes_output/WiDiv510_PH207_chr4B_gene1kb.vcf -V ~/freebayes_output/WiDiv510_PH207_chr4C_gene1kb.vcf -V ~/freebayes_output/WiDiv510_PH207_chr5A_gene1kb.vcf -V ~/freebayes_output/WiDiv510_PH207_chr5B_gene1kb.vcf -V ~/freebayes_output/WiDiv510_PH207_chr5C_gene1kb.vcf -V ~/freebayes_output/WiDiv510_PH207_chr6A_gene1kb.vcf -V ~/freebayes_output/WiDiv510_PH207_chr6B_gene1kb.vcf  -V ~/freebayes_output/WiDiv510_PH207_chr7A_gene1kb.vcf -V ~/freebayes_output/WiDiv510_PH207_chr7B_gene1kb.vcf -V ~/freebayes_output/WiDiv510_PH207_chr8A_gene1kb.vcf -V ~/freebayes_output/WiDiv510_PH207_chr8B_gene1kb.vcf -V ~/freebayes_output/WiDiv510_PH207_chr9A_gene1kb.vcf -V ~/freebayes_output/WiDiv510_PH207_chr9B_gene1kb.vcf -V ~/freebayes_output/WiDiv510_PH207_chr10A_gene1kb.vcf -V ~/freebayes_output/WiDiv510_PH207_chr10B_gene1kb.vcf -out ~/freebayes_output/WiDiv510_PH207_allchr_gene1kb.vcf 

#java -cp GenomeAnalysisTK.jar org.broadinstitute.gatk.tools.CatVariants -R ~/maize_refs/W22_chr1-10.fasta -V ~/freebayes_output/WiDiv509_W22v12_chr1A_gene1kb.vcf -V ~/freebayes_output/WiDiv509_W22v12_chr1B_gene1kb.vcf -V ~/freebayes_output/WiDiv509_W22v12_chr1C_gene1kb.vcf -V ~/freebayes_output/WiDiv509_W22v12_chr1D_gene1kb.vcf -V ~/freebayes_output/WiDiv509_W22v12_chr2A_gene1kb.vcf -V ~/freebayes_output/WiDiv509_W22v12_chr2B_gene1kb.vcf -V ~/freebayes_output/WiDiv509_W22v12_chr2C_gene1kb.vcf -V ~/freebayes_output/WiDiv509_W22v12_chr3A_gene1kb.vcf -V ~/freebayes_output/WiDiv509_W22v12_chr3B_gene1kb.vcf -V ~/freebayes_output/WiDiv509_W22v12_chr3C_gene1kb.vcf -V ~/freebayes_output/WiDiv509_W22v12_chr4A_gene1kb.vcf -V ~/freebayes_output/WiDiv509_W22v12_chr4B_gene1kb.vcf -V ~/freebayes_output/WiDiv509_W22v12_chr4C_gene1kb.vcf -V ~/freebayes_output/WiDiv509_W22v12_chr5A_gene1kb.vcf -V ~/freebayes_output/WiDiv509_W22v12_chr5B_gene1kb.vcf -V ~/freebayes_output/WiDiv509_W22v12_chr5C_gene1kb.vcf -V ~/freebayes_output/WiDiv509_W22v12_chr6A_gene1kb.vcf -V ~/freebayes_output/WiDiv509_W22v12_chr6B_gene1kb.vcf  -V ~/freebayes_output/WiDiv509_W22v12_chr7A_gene1kb.vcf -V ~/freebayes_output/WiDiv509_W22v12_chr7B_gene1kb.vcf -V ~/freebayes_output/WiDiv509_W22v12_chr8A_gene1kb.vcf -V ~/freebayes_output/WiDiv509_W22v12_chr8B_gene1kb.vcf -V ~/freebayes_output/WiDiv509_W22v12_chr9A_gene1kb.vcf -V ~/freebayes_output/WiDiv509_W22v12_chr9B_gene1kb.vcf -V ~/freebayes_output/WiDiv509_W22v12_chr10A_gene1kb.vcf -V ~/freebayes_output/WiDiv509_W22v12_chr10B_gene1kb.vcf -out ~/freebayes_output/WiDiv509_W22v12_allchr_gene1kb_run2.vcf 


#SNP and indel filtering 
#cd /home/hirschc1/oconnorc/freebayes_output

#mv WiDiv509_W22v12_chr*vcf freebayes_indvl_chr/
#mv WiDiv510_PH207_chr*vcf freebayes_indvl_chr/
#mv WiDiv512_Mo17_chr*vcf freebayes_indvl_chr/


#select correct 509 genotypes and only indels >= 2bp 
#gatk SelectVariants -R /home/hirschc1/oconnorc/maize_refs/Mo17_chr1-10.fasta -select-type INDEL --sample-name /home/hirschc1/oconnorc/freebayes_output/Mo17_sample_names.args -V /home/hirschc1/oconnorc/freebayes_output/WiDiv512_Mo17_allchr_gene1kb.vcf --min-indel-size 2 -O /home/hirschc1/oconnorc/freebayes_output/WiDiv508_Mo17_allchr_indels.vcf
#gatk SelectVariants -R /home/hirschc1/oconnorc/maize_refs/Mo17_chr1-10.fasta -select-type INDEL --sample-name /home/hirschc1/oconnorc/freebayes_output/PHB47_sample_names.args -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_allchr_gene1kb.vcf --min-indel-size 2 -O /home/hirschc1/oconnorc/freebayes_output/WiDiv508_PHB47_allchr_indels.vcf

#gatk SelectVariants -R ~/maize_refs/PH207_chr1-10.fasta -select-type INDEL --sample-name ~/freebayes_output/PH207_sample_names.args -V ~/freebayes_output/WiDiv510_PH207_allchr_gene1kb.vcf --min-indel-size 2 -O ~/freebayes_output/WiDiv508_PH207_allchr_indels.vcf
#gatk SelectVariants -R ~/maize_refs/W22_chr1-10.fasta -select-type INDEL --sample-name ~/freebayes_output/W22_sample_names.args -V ~/freebayes_output/WiDiv509_W22v12_allchr_gene1kb.vcf --min-indel-size 2 -O ~/freebayes_output/WiDiv508_W22v12_allchr_indels.vcf

#select SNPs 
#gatk SelectVariants -R /home/hirschc1/oconnorc/maize_refs/Mo17_chr1-10.fasta -select-type SNP --sample-name /home/hirschc1/oconnorc/freebayes_output/Mo17_sample_names.args -V /home/hirschc1/oconnorc/freebayes_output/WiDiv512_Mo17_allchr_gene1kb.vcf -O /home/hirschc1/oconnorc/freebayes_output/WiDiv508_Mo17_allchr_SNPs.vcf
#gatk VariantFiltration -R /home/hirschc1/oconnorc/maize_refs/Mo17_chr1-10.fasta -V /home/hirschc1/oconnorc/freebayes_output/WiDiv508_Mo17_allchr_SNPs.vcf --filter-expression "QD < 2.0 || FS > 60.0 || MQ < 40.0 || MQRankSum < -12.5 || ReadPosRankSum < -8.0" --filter-name "GATK_snp_filter" -O /home/hirschc1/oconnorc/freebayes_output/WiDiv508_Mo17_allchr_filtered_SNPs.vcf #/home/hirschc1/oconnorc/software/vcftools_0.1.13/bin/vcftools --vcf /home/hirschc1/oconnorc/freebayes_output/WiDiv508_Mo17_allchr_filtered_SNPs.vcf  --remove-filtered-all --max-missing 0.10 --mac 50 --minQ 30 --recode --recode-INFO-all --out /home/hirschc1/oconnorc/freebayes_output/WiDiv508_Mo17_allchr_SNPS_maxmiss0.10

#gatk SelectVariants -R /home/hirschc1/oconnorc/maize_refs/PHB47_chr1-10.fasta -select-type SNP --sample-name /home/hirschc1/oconnorc/freebayes_output/PHB47_sample_names.args -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_allchr_gene1kb.vcf -O /home/hirschc1/oconnorc/freebayes_output/WiDiv508_PHB47_allchr_SNPs.vcf
#gatk VariantFiltration -R /home/hirschc1/oconnorc/maize_refs/PHB47_chr1-10.fasta -V /home/hirschc1/oconnorc/freebayes_output/WiDiv508_PHB47_allchr_SNPs.vcf --filter-expression "QD < 2.0 || FS > 60.0 || MQ < 40.0 || MQRankSum < -12.5 || ReadPosRankSum < -8.0" --filter-name "GATK_snp_filter" -O /home/hirschc1/oconnorc/freebayes_output/WiDiv508_PHB47_allchr_filtered_SNPs.vcf
#/home/hirschc1/oconnorc/software/vcftools_0.1.13/bin/vcftools --vcf /home/hirschc1/oconnorc/freebayes_output/WiDiv508_PHB47_allchr_filtered_SNPs.vcf  --remove-filtered-all --max-missing 0.10 --mac 50 --minQ 30 --recode --recode-INFO-all --out /home/hirschc1/oconnorc/freebayes_output/WiDiv508_PHB47_allchr_SNPS_maxmiss0.10

#gatk SelectVariants -R ~/maize_refs/PH207_chr1-10.fasta -select-type SNP --sample-name ~/freebayes_output/PH207_sample_names.args -V ~/freebayes_output/WiDiv510_PH207_allchr_gene1kb.vcf -O ~/freebayes_output/WiDiv508_PH207_allchr_SNPs.vcf
#gatk VariantFiltration -R ~/maize_refs/PH207_chr1-10.fasta -V ~/freebayes_output/WiDiv508_PH207_allchr_SNPs.vcf --filter-expression "QD < 2.0 || FS > 60.0 || MQ < 40.0 || MQRankSum < -12.5 || ReadPosRankSum < -8.0" --filter-name "GATK_snp_filter" -O ~/freebayes_output/WiDiv508_PH207_allchr_filtered_SNPs.vcf 
#~/software/vcftools_0.1.13/bin/vcftools --vcf ~/freebayes_output/WiDiv508_PH207_allchr_filtered_SNPs.vcf  --remove-filtered-all --max-missing 0.10 --mac 50 --minQ 30 --recode --recode-INFO-all --out ~/freebayes_output/WiDiv508_PH207_allchr_SNPS_maxmiss0.10

#gatk SelectVariants -R ~/maize_refs/W22_chr1-10.fasta -select-type SNP --sample-name ~/freebayes_output/W22_sample_names.args -V ~/freebayes_output/WiDiv509_W22v12_allchr_gene1kb.vcf -O ~/freebayes_output/WiDiv508_W22v12_allchr_SNPs.vcf
#gatk VariantFiltration -R ~/maize_refs/W22_chr1-10.fasta -V ~/freebayes_output/WiDiv508_W22v12_allchr_SNPs.vcf --filter-expression "QD < 2.0 || FS > 60.0 || MQ < 40.0 || MQRankSum < -12.5 || ReadPosRankSum < -8.0" --filter-name "GATK_snp_filter" -O ~/freebayes_output/WiDiv508_W22v12_allchr_filtered_SNPs.vcf 
#~/software/vcftools_0.1.13/bin/vcftools --vcf ~/freebayes_output/WiDiv508_W22v12_allchr_filtered_SNPs.vcf  --remove-filtered-all --max-missing 0.10 --mac 50 --minQ 30 --recode --recode-INFO-all --out ~/freebayes_output/WiDiv508_W22v12_allchr_SNPS_maxmiss0.10


cd /home/hirschc1/oconnorc/freebayes_output
#fix freebayes indel call files 
#re-do PHB47 because reference genome for indel SelectVariants is incorrect 
gatk SelectVariants -R /home/hirschc1/oconnorc/maize_refs/PHB47_chr1-10.fasta -select-type INDEL --sample-name /home/hirschc1/oconnorc/freebayes_output/PHB47_sample_names.args -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_allchr_gene1kb.vcf --min-indel-size 2 -O /home/hirschc1/oconnorc/freebayes_output/WiDiv508_PHB47_allchr_indels.vcf
gatk SelectVariants -R /home/hirschc1/oconnorc/maize_refs/W22_chr1-10.fasta -select-type INDEL --sample-name /home/hirschc1/oconnorc/freebayes_output/W22_sample_names.args -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_W22v12_allchr_gene1kb_run2.vcf --min-indel-size 2 -O /home/hirschc1/oconnorc/freebayes_output/WiDiv508_W22v12_allchr_indels.vcf

#keep all 509 genotypes for all samples 
gatk SelectVariants -R /home/hirschc1/oconnorc/maize_refs/B73_chr1-10.fasta -select-type INDEL --sample-name /home/hirschc1/oconnorc/freebayes_output/WiDiv509_samplenames.args -V /home/hirschc1/oconnorc/freebayes_output/WiDiv512_B73_allchr_gene1kb.vcf --min-indel-size 2 -O /home/hirschc1/oconnorc/freebayes_output/WiDiv509_Mo17_fb_allchr_indels.vcf
gatk SelectVariants -R /home/hirschc1/oconnorc/maize_refs/Mo17_chr1-10.fasta -select-type INDEL --sample-name /home/hirschc1/oconnorc/freebayes_output/WiDiv509_samplenames.args -V /home/hirschc1/oconnorc/freebayes_output/WiDiv512_Mo17_allchr_gene1kb.vcf --min-indel-size 2 -O /home/hirschc1/oconnorc/freebayes_output/WiDiv509_Mo17_fb_allchr_indels.vcf
gatk SelectVariants -R /home/hirschc1/oconnorc/maize_refs/PHB47_chr1-10.fasta -select-type INDEL --sample-name /home/hirschc1/oconnorc/freebayes_output/WiDiv509_samplenames.args -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_allchr_gene1kb.vcf --min-indel-size 2 -O /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_fb_allchr_indels.vcf
gatk SelectVariants -R /home/hirschc1/oconnorc/maize_refs/PH207_chr1-10.fasta -select-type INDEL --sample-name /home/hirschc1/oconnorc/freebayes_output/WiDiv509_samplenames.args -V /home/hirschc1/oconnorc/freebayes_output/WiDiv510_PH207_allchr_gene1kb.vcf --min-indel-size 2 -O /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PH207_fb_allchr_indels.vcf
gatk SelectVariants -R /home/hirschc1/oconnorc/maize_refs/W22_chr1-10.fasta -select-type INDEL --sample-name /home/hirschc1/oconnorc/freebayes_output/WiDiv509_samplenames.args -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_W22v12_allchr_gene1kb.vcf --min-indel-size 2 -O /home/hirschc1/oconnorc/freebayes_output/WiDiv509_W22v12_fb_allchr_indels.vcf


#re-do SNP filters 
#submitted: 06/22/2020
gatk SelectVariants -R /home/hirschc1/oconnorc/maize_refs/B73_chr1-10.fasta -select-type SNP --sample-name /home/hirschc1/oconnorc/freebayes_output/WiDiv509_samplenames.args -V /home/hirschc1/oconnorc/freebayes_output/WiDiv512_B73_allchr_gene1kb.vcf -O /home/hirschc1/oconnorc/freebayes_output/WiDiv509_B73v4_allchr_SNPs.vcf
gatk VariantFiltration -R /home/hirschc1/oconnorc/maize_refs/B73_chr1-10.fasta -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_B73v4_allchr_SNPs.vcf --filter-expression "QD < 2.0 || FS > 60.0 || MQ < 40.0 || MQRankSum < -12.5 || ReadPosRankSum < -8.0" --filter-name "GATK_snp_filter" -O /home/hirschc1/oconnorc/freebayes_output/WiDiv509_B73v4_allchr_filtered_SNPs.vcf
/home/hirschc1/oconnorc/software/vcftools_0.1.13/bin/vcftools --vcf /home/hirschc1/oconnorc/freebayes_output/WiDiv509_B73v4_allchr_filtered_SNPs.vcf  --remove-filtered-all --max-missing 0.10 --mac 50 --minQ 30 --recode --recode-INFO-all --out /home/hirschc1/oconnorc/freebayes_output/WiDiv509_B73v4_allchr_SNPS_maxmiss0.10

gatk SelectVariants -R /home/hirschc1/oconnorc/maize_refs/Mo17_chr1-10.fasta -select-type SNP --sample-name /home/hirschc1/oconnorc/freebayes_output/WiDiv509_samplenames.args -V /home/hirschc1/oconnorc/freebayes_output/WiDiv512_Mo17_allchr_gene1kb.vcf -O /home/hirschc1/oconnorc/freebayes_output/WiDiv509_Mo17_allchr_SNPs.vcf
gatk VariantFiltration -R /home/hirschc1/oconnorc/maize_refs/Mo17_chr1-10.fasta -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_Mo17_allchr_SNPs.vcf --filter-expression "QD < 2.0 || FS > 60.0 || MQ < 40.0 || MQRankSum < -12.5 || ReadPosRankSum < -8.0" --filter-name "GATK_snp_filter" -O /home/hirschc1/oconnorc/freebayes_output/WiDiv509_Mo17_allchr_filtered_SNPs.vcf
/home/hirschc1/oconnorc/software/vcftools_0.1.13/bin/vcftools --vcf /home/hirschc1/oconnorc/freebayes_output/WiDiv509_Mo17_allchr_filtered_SNPs.vcf  --remove-filtered-all --max-missing 0.10 --mac 50 --minQ 30 --recode --recode-INFO-all --out /home/hirschc1/oconnorc/freebayes_output/WiDiv509_Mo17_allchr_SNPS_maxmiss0.10

gatk SelectVariants -R /home/hirschc1/oconnorc/maize_refs/PHB47_chr1-10.fasta -select-type SNP --sample-name /home/hirschc1/oconnorc/freebayes_output/WiDiv509_samplenames.args -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_allchr_gene1kb.vcf -O /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_allchr_SNPs.vcf
gatk VariantFiltration -R /home/hirschc1/oconnorc/maize_refs/PHB47_chr1-10.fasta -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_allchr_SNPs.vcf --filter-expression "QD < 2.0 || FS > 60.0 || MQ < 40.0 || MQRankSum < -12.5 || ReadPosRankSum < -8.0" --filter-name "GATK_snp_filter" -O /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_allchr_filtered_SNPs.vcf
/home/hirschc1/oconnorc/software/vcftools_0.1.13/bin/vcftools --vcf /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_allchr_filtered_SNPs.vcf  --remove-filtered-all --max-missing 0.10 --mac 50 --minQ 30 --recode --recode-INFO-all --out /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PHB47_allchr_SNPS_maxmiss0.10

gatk SelectVariants -R /home/hirschc1/oconnorc/maize_refs/PH207_chr1-10.fasta -select-type SNP --sample-name /home/hirschc1/oconnorc/freebayes_output/WiDiv509_samplenames.args -V /home/hirschc1/oconnorc/freebayes_output/WiDiv510_PH207_allchr_gene1kb.vcf -O ~/freebayes_output/WiDiv509_PH207_allchr_SNPs.vcf
gatk VariantFiltration -R /home/hirschc1/oconnorc/maize_refs/PH207_chr1-10.fasta -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PH207_allchr_SNPs.vcf --filter-expression "QD < 2.0 || FS > 60.0 || MQ < 40.0 || MQRankSum < -12.5 || ReadPosRankSum < -8.0" --filter-name "GATK_snp_filter" -O /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PH207_allchr_filtered_SNPs.vcf 
/home/hirschc1/oconnorc/software/vcftools_0.1.13/bin/vcftools --vcf /home/hirschc1/oconnorc/freebayes_output/WiDiv509_PH207_allchr_filtered_SNPs.vcf  --remove-filtered-all --max-missing 0.10 --mac 50 --minQ 30 --recode --recode-INFO-all --out ~/freebayes_output/WiDiv509_PH207_allchr_SNPS_maxmiss0.10

gatk SelectVariants -R /home/hirschc1/oconnorc/maize_refs/W22_chr1-10.fasta -select-type SNP --sample-name /home/hirschc1/oconnorc/freebayes_output/WiDiv509_samplenames.args -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_W22v12_allchr_gene1kb_run2.vcf -O ~/freebayes_output/WiDiv509_W22v12_allchr_SNPs.vcf
gatk VariantFiltration -R /home/hirschc1/oconnorc/maize_refs/W22_chr1-10.fasta -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_W22v12_allchr_SNPs.vcf --filter-expression "QD < 2.0 || FS > 60.0 || MQ < 40.0 || MQRankSum < -12.5 || ReadPosRankSum < -8.0" --filter-name "GATK_snp_filter" -O /home/hirschc1/oconnorc/freebayes_output/WiDiv509_W22v12_allchr_filtered_SNPs.vcf 
/home/hirschc1/oconnorc/software/vcftools_0.1.13/bin/vcftools --vcf /home/hirschc1/oconnorc/freebayes_output/WiDiv509_W22v12_allchr_filtered_SNPs.vcf  --remove-filtered-all --max-missing 0.10 --mac 50 --minQ 30 --recode --recode-INFO-all --out ~/freebayes_output/WiDiv509_W22v12_allchr_SNPS_maxmiss0.10

#re-do W22
gatk SelectVariants -R /home/hirschc1/oconnorc/maize_refs/W22_chr1-10.fasta -select-type SNP --sample-name /home/hirschc1/oconnorc/freebayes_output/W22_sample_names.args -V /home/hirschc1/oconnorc/freebayes_output/WiDiv509_W22v12_allchr_gene1kb_run2.vcf -O /home/hirschc1/oconnorc/freebayes_output/WiDiv508_W22v12_allchr_SNPs.vcf
gatk VariantFiltration -R /home/hirschc1/oconnorc/maize_refs/W22_chr1-10.fasta -V /home/hirschc1/oconnorc/freebayes_output/WiDiv508_W22v12_allchr_SNPs.vcf --filter-expression "QD < 2.0 || FS > 60.0 || MQ < 40.0 || MQRankSum < -12.5 || ReadPosRankSum < -8.0" --filter-name "GATK_snp_filter" -O /home/hirschc1/oconnorc/freebayes_output/WiDiv508_W22v12_allchr_filtered_SNPs.vcf
/home/hirschc1/oconnorc/software/vcftools_0.1.13/bin/vcftools --vcf /home/hirschc1/oconnorc/freebayes_output/WiDiv508_W22v12_allchr_filtered_SNPs.vcf  --remove-filtered-all --max-missing 0.10 --mac 50 --minQ 30 --recode --recode-INFO-all --out /home/hirschc1/oconnorc/freebayes_output/WiDiv508_W22v12_allchr_SNPS_maxmiss0.10

#end of script

